### ISSUE TITLE
Possible improvement to generic-elt on objects

### ISSUE THREAD

**Issue by jasom** [vseloved/rutils#28]
From your blog post:

> Still, one issue remains unsolved in this approach: the preferred Lisp slot-access method is not via slot-value, but with an accessor method that is exported

given that the preferred method is with an accessor, why not allow something like `(? obj #'accessor-fn)`

**Comment by vseloved** [comment]
Yes, this sounds reasonable. I'll add this in the next update. Also, patches are welcome :)

**Comment by jcguu95** [comment]
Unfortunately, I think this is not possible unless we break API. One counter example is

``` lisp
(let ((accessor (lambda (ht) (gethash :a ht)))) 
  (? #h(accessor 0 :a 1) accessor))
```

If the it works in the new semantics, it should return `1`. However, in the old semantics it returns `0`.

I suggest to close this issue.

**Comment by jasom** [comment]
I totally spaced on this, but wouldn't this be solved by requiring it to be a `cl:function` form as it was in my initial comment?  It does mean there's no dynamic dispatch on accessors though...
